package/MDAnalysis/coordinates/TRR.py

```diff
@@ -1,123 +1,157 @@
-          ts.time = frame-1
-       else:
-          ts.time = mode.frequency
-       W.write(ts)             # converts angstrom to nm for gmx
-
-    W.close()
-
-.. _MMTK: http://dirac.cnrs-orleans.fr/Manuals/MMTK/index.html
-
-.. _`recipe by Ramon Crehuet`: https://github.com/MDAnalysis/mdanalysis/issues/79
-
-Module reference
-----------------
-
-.. autoclass:: Timestep
-   :members:
-   :inherited-members:
-
-.. autoclass:: TRRReader
-   :members:
-   :inherited-members:
-
-.. autoclass:: TRRWriter
-   :members:
-   :inherited-members:
-"""
-
-from .xdrfile.TRR import TRRReader, TRRWriter, Timestep
+import errno
+
+from . import base
+from ..lib.formats.trr import TRRFile
+from ..lib.mdamath import triclinic_box, triclinic_vectors
+
+
+class TRRReader(base.Reader):
+    format = 'TRR'
+    units = {'time': 'ps', 'length': 'nm', 'velocity': 'nm/ps',
+             'force': 'kJ/(mol*nm)'}
+
+    def __init__(self, filename, convert_units=True, sub=None, **kwargs):
+        super(TRRReader, self).__init__(filename, convert_units=convert_units,
+                                        **kwargs)
+        self._trr = TRRFile(filename)
+
+        self.n_frames = len(self._trr)
+        self._sub = sub
+        trr_frame = self._trr.read()
+        try:
+            trr_frame_2 = self._trr.read()
+            time_2 = trr_frame_2.time
+        except:
+            time_2 = trr_frame.time
+
+        if sub is not None:
+            self.n_atoms = len(sub)
+        else:
+            self.n_atoms = self._trr.n_atoms
+        self._trr.seek(0)
+
+        self.ts = self._Timestep(self.n_atoms, **self._ts_kwargs)
+        self._frame = 0
+        self.ts = self.frame_to_ts(trr_frame, self.ts)
+
+        self.ts.dt = time_2 - trr_frame.time
+        self.ts.dimensions = triclinic_box(*trr_frame.box)
+
+        if self.convert_units:
+            self.convert_pos_from_native(self.ts.dimensions[:3])
+
+        self.rewind()
+
+    def frame_to_ts(self, frame, ts):
+        ts.has_positions = frame.hasx
+        ts.has_velocities = frame.hasv
+        ts.has_forces = frame.hasf
+
+        if ts.has_positions:
+            if self._sub is not None:
+                ts._pos = frame.x[self._sub]
+            else:
+                ts._pos = frame.x
+            if self.convert_units:
+                self.convert_pos_from_native(ts._pos)
+
+        if ts.has_velocities:
+            if self._sub is not None:
+                ts.velocities = frame.v[self._sub]
+            else:
+                ts.velocities = frame.v
+            if self.convert_units:
+                self.convert_velocities_from_native(ts.velocities)
+
+        if ts.has_forces:
+            if self._sub is not None:
+                ts.forces = frame.f[self._sub]
+            else:
+                ts.forces = frame.f
+            if self.convert_units:
+                self.convert_forces_from_native(ts.forces)
+
+        ts.time = frame.time
+        ts.frame = self._frame
```

Comment:
But you'll need to update frame **first**, because that's how the whole system, works, it records which frame the data came from, then compares current frame to that frame when accessing.